internal/advtools/convo.lua

```diff
@@ -61,8 +61,111 @@ local function generateKeywordsForChoice(choice)
+    choice.cc = df.talk_choice:new()
+    choice.cc.type = choice_type
+    local text = df.new("string")
+    text.value = title
+    choice.print_string.text:insert("#", text)
+
+    if keywords ~= nil then
+        addKeywords(choice, keywords)
+    end
+    return choice
+end
+
+local function addHistFigWhereaboutsChoice(profile)
+    for i, c in pairs(adventure.conversation.conv_choice_info) do
+        if c.cc.type == df.talk_choice_type.AskWhereabouts and
+          c.cc.invocation_target_hfid ~= -1 and
+          c.cc.invocation_target_hfid == profile.histfig_id then
+            -- Don't add repeat entries
+            return
+        end
+    end
+
+    local histfig = df.historical_figure.find(profile.histfig_id)
+    local name = ""
+    local creature = df.creature_raw.find(histfig.race)
+    if creature then
+        local caste = creature.caste[histfig.caste]
+        name = caste.caste_name[0]
+    end
+    local title = "Ask for the whereabouts of the " .. name .. " " .. dfhack.TranslateName(histfig.name, true)
+    if profile._type == df.relationship_profile_hf_historicalst then
+        title = title .. " (Heard of)"
+    end
+    local choice = new_choice(df.talk_choice_type.AskWhereabouts, title, dfhack.TranslateName(histfig.name):split())
+     -- insert before the last choice, which is usually "back"
+    adventure.conversation.conv_choice_info:insert(#adventure.conversation.conv_choice_info-1, choice)
+    choice.cc.invocation_target_hfid = histfig.id
+end
+
+local function addIdentityWhereaboutsChoice(identity)
+    for i, c in pairs(adventure.conversation.conv_choice_info) do
+        if c.cc.type == df.talk_choice_type.AskWhereabouts and
+           c.cc.invocation_target_hfid ~= -1 and
+           c.cc.invocation_target_hfid == identity.impersonated_hf then
+            -- Don't add repeat entries
+            return
+        end
+    end
+    local identity_name = identity.name
+    local name = ""
+    local creature = df.creature_raw.find(identity.race)
+    if creature then
+        local caste = creature.caste[identity.caste]
+        name = caste.caste_name[0]
+    else
+        -- no race given for the identity, assume it's the histfig
+        local histfig = df.historical_figure.find(identity.histfig_id)
+        creature = df.creature_raw.find(histfig.race)
+        if creature then
+            local caste = creature.caste[histfig.caste]
+            name = caste.caste_name[0]
+        end
+    end
+    local title = "Ask for the whereabouts of the " .. name .. " " .. dfhack.TranslateName(identity_name, true)
+    local choice = new_choice(df.talk_choice_type.AskWhereabouts, title)
+    -- insert before the last choice, which is usually "back"
+    adventure.conversation.conv_choice_info:insert(#adventure.conversation.conv_choice_info-1, choice, dfhack.TranslateName(identity_name):split())
+    choice.cc.invocation_target_hfid = identity.impersonated_hf
+end
```

Comment:
this duplicates too much of the logic in `addHistFigWhereaboutsChoice`; please refactor